### DISCUSSION TITLE
PnP Modern - how to exclude or select certain columns

### DISCUSSION THREAD

**Discussion by mattyjacksy** [microsoft-search/pnp-modern-search#3236] [Q&A] [answered]
I have defined 8 columns (Authors, Year, Journal, Topics etc.) and done all the mapping. The PnP works. But when I search my own name I get all files from the searched document library. I assume all documents CreatedBy is being search. I've searched and tried multiple ways.

Is there a way to simply limit the search to the 8 created columns via the Query template. 

The Query template I have currently maps to a Path:sharepoint.com/sites/team/library with IsDocument:True. I assume I can built or add a restrict to columns, but any help with this would be appreciated!

**Comment by kasperbolarsen** [comment, marked as answer]
Are you looking for something like Author:{searchTerms} OR Year:{searchTerms} and so on?

**Reply by mattyjacksy** [reply]
thanks for the quick response! Exactly what was needed.

**Reply by wobba** [reply]
And it's better to add some () in the template to ensure KQL logic as intended when more than one term is entered: `Author:({searchTerms}) OR Year:({searchTerms})`
